### PR TITLE
feat: support plugin config

### PR DESCRIPTION
**Pull request by nic-chen**
Please answer these questions before submitting a pull request

- Why submit this pull request?
- [ ] Bugfix
- [x] New feature provided
- [ ] Improve performance
- [ ] Backport patches

support plugin config in Manager API
related pr in APISIX https://github.com/apache/apisix/pull/3567

### CODE REVIEW

File: api/internal/core/store/storehub.go [hunk 1]
```diff
@@ -36,6 +36,7 @@ const (
 	HubKeyScript     HubKey = "script"
 	HubKeyGlobalRule HubKey = "global_rule"
 	HubKeyServerInfo HubKey = "server_info"
+	HubKeyPluginConfig   HubKey = "plugin_config"
```
Comment (starsz) [inline api/internal/core/store/storehub.go:39]: Code style.

File: api/internal/handler/plugin_config/plugin_config.go [hunk 2]
```diff
@@ -0,0 +1,228 @@
...
+	// check if ID in body is equal ID in path
+	if err := handler.IDCompare(input.ID, input.PluginConfig.ID); err != nil {
+		return &data.SpecCodeResponse{StatusCode: http.StatusBadRequest}, err
+	}
+
+	if input.ID != "" {
+		input.PluginConfig.ID = input.ID
+	}
+
+	ret, err := h.pluginConfigStore.Update(c.Context(), &input.PluginConfig, true)
```
Comment (starsz) [inline api/internal/handler/plugin_config/plugin_config.go:169]: Please make sure about the third args in `h.pluginConfigStore.Update`.
We should create it if not exist?

File: api/internal/handler/plugin_config/plugin_config.go [hunk 3]
```diff
@@ -0,0 +1,228 @@
...
+type BatchDelete struct {
+	IDs string `auto_read:"ids,path"`
+}
+
+func (h *Handler) BatchDelete(c droplet.Context) (interface{}, error) {
+	input := c.Input().(*BatchDelete)
+
+	if err := h.pluginConfigStore.BatchDelete(c.Context(), strings.Split(input.IDs, ",")); err != nil {
+		return handler.SpecCodeResponse(err), err
+	}
```
Comment (starsz) [inline api/internal/handler/plugin_config/plugin_config.go:186]: IMO, we need to check whether it is used when deleting it.

File: api/internal/handler/plugin_config/plugin_config.go [hunk 4]
```diff
@@ -0,0 +1,228 @@
...
+
+	var pluginConfig entity.PluginConfig
+	if err := json.Unmarshal(res, &pluginConfig); err != nil {
+		log.Warnf("%s", err)
+		return handler.SpecCodeResponse(err), err
+	}
+
+	ret, err := h.pluginConfigStore.Update(c.Context(), &pluginConfig, false)
+	if err != nil {
+		log.Warnf("%s", err)
```
Comment (starsz) [inline api/internal/handler/plugin_config/plugin_config.go:223]: The log seems not very good.

File: api/internal/handler/plugin_config/plugin_config.go (hunk 2 above)
Comment (nic-chen) [reply api/internal/handler/plugin_config/plugin_config.go:169]: > Please make sure about the third args in `h.pluginConfigStore.Update`.
> We should create it if not exist?

sure, we support create by `PUT`

File: api/internal/core/store/storehub.go (hunk 1 above)
Comment (nic-chen) [reply api/internal/core/store/storehub.go]: fixed.

File: api/internal/handler/plugin_config/plugin_config.go (hunk 3 above)
Comment (nic-chen) [reply api/internal/handler/plugin_config/plugin_config.go:214]: updated.

File: api/internal/handler/plugin_config/plugin_config.go (hunk 4 above)
Comment (nic-chen) [reply api/internal/handler/plugin_config/plugin_config.go]: fixed.

File: api/internal/handler/plugin_config/plugin_config.go (hunk 2 above)
Comment (starsz) [reply api/internal/handler/plugin_config/plugin_config.go:172]: OK.Got it.

File: api/test/e2enew/plugin_config/plugin_config_test.go [hunk 5]
```diff
@@ -0,0 +1,189 @@
...
+					"nodes": [{
+						"host": "` + base.UpstreamIp + `",
+						"port": 1981,
+						"weight": 1
+					}]
+				 }
+			 }`,
+			Headers:      map[string]string{"Authorization": base.GetToken()},
+			ExpectStatus: http.StatusOK,
+		}),
```
Comment (starsz) [inline api/test/e2enew/plugin_config/plugin_config_test.go:77]: It seems missing `get plugin-config?

Comment (nic-chen) [reply api/test/e2enew/plugin_config/plugin_config_test.go:85]: fixed.

File: api/test/e2enew/plugin_config/plugin_config_test.go [hunk 6]
```diff
@@ -0,0 +1,197 @@
...
+					},
+					"uri-blocker": {
+						"block_rules": ["select.+(from|limit)", "(?:(union(.*?)select))"]
+					}
+				}
+			}`,
+			Headers:      map[string]string{"Authorization": base.GetToken()},
+			ExpectStatus: http.StatusOK,
+		}),
+		table.Entry("create plugin config", base.HttpTestCase{
```
Comment (starsz) [inline api/test/e2enew/plugin_config/plugin_config_test.go:59]: It seems `create plugin config` => `get plugin config`.

File: api/test/e2enew/plugin_config/plugin_config_test.go [hunk 7]
```diff
@@ -0,0 +1,197 @@
...
+			Object: base.ManagerApiExpect(),
+			Path:   "/apisix/admin/plugin_configs/1",
```
Comment (starsz) [inline api/test/e2enew/plugin_config/plugin_config_test.go:61]: Need to test `list` function.

**Comment by gxthrj** [comment]
E2E test failed

File: api/test/e2enew/plugin_config/plugin_config_test.go (hunk 6 above)
Comment (nic-chen) [reply api/test/e2enew/plugin_config/plugin_config_test.go]: fixed.

File: api/test/e2enew/plugin_config/plugin_config_test.go (hunk 7 above)
Comment (nic-chen) [reply api/test/e2enew/plugin_config/plugin_config_test.go]: fixed.

File: api/test/e2e/label_test.go [hunk 8]
```diff
@@ -302,6 +333,14 @@ func TestLabel(t *testing.T) {
 			Headers:      map[string]string{"Authorization": token},
 			ExpectStatus: http.StatusOK,
 		},
+		{
```
Comment (tokers) [inline api/test/e2e/label_test.go:336]: Can `plugin_config` be updated? If so, we should also add update test suites.

**Review by starsz — APPROVED** [review]

File: api/test/e2e/label_test.go (hunk 8 above)
Comment (nic-chen) [reply api/test/e2e/label_test.go:367]: updated.

**Review by tokers — APPROVED** [review]

**Pull request merged**